tensorflow/python/ops/math_ops.py

```diff
@@ -99,9 +99,49 @@
 from tensorflow.python.util.tf_export import tf_export
 
 # Aliases for some automatically-generated names.
-linspace = gen_math_ops.lin_space
+#linspace = gen_math_ops.lin_space
 nextafter = gen_math_ops.next_after
 
+def linspace(start_in, stop_in, num, axis=0, name=None):
+  if not isinstance(axis, int):
+    raise TypeError('Axis should be an integer. Received an object {0} of type {1}'.format(axis, type(axis)))
+  with ops.name_scope(name, 'linspace', [start_in, stop_in]):
+    start = ops.convert_to_tensor(start_in, name='start')
+    # stop must be convertible to the same dtype as start
+    stop = ops.convert_to_tensor(stop_in, name='stop', dtype=start.dtype)
+    expanded_start = array_ops.expand_dims(start, axis=axis)
+    num_int = array_ops.convert_to_int_tensor(num, name='num')
+    num = cast(num_int, dtype=start.dtype)
+    shape = expanded_start.get_shape()
+    axis = array_ops.get_positive_axis(axis, shape.ndims)
+    expanded_stop = array_ops.expand_dims(stop, axis=axis)
+    # to avoid having negative values in the range or zero division
+    # The result is sliced in the end so a correct result is returned for num == 1.
+    range_end = gen_math_ops.maximum(num - 1., 1.)
+    delta = (expanded_stop - expanded_start) / range_end
+    shape_range = range(shape.ndims)
+    axis_tiled = array_ops.fill(shape_range.get_shape(), axis)
```

Comment:
Why not tf.ones_like(shape_range) * axis?

Reply:
Your proposal is indeed more elegant. Moreover, I think that with the `tf.ones_like` I am going to be able to handle `None` shapes as well (similar to one of your previous comments). I have changed this below, let me know if it's still an issue.